tests/parse-legacy-file.bats

```diff
@@ -0,0 +1,161 @@
+
+  echo "${expected_opentofu_version}" >"${tmpdir}/.opentofu-version"
+
+  local -r actual_opentofu_version="$(ASDF_OPENTOFU_THIS_PLUGIN=opentofu "${PARSE_LEGACY_FILE}" "${version_file}")"
+
+  [[ ${actual_opentofu_version} == "${expected_opentofu_version}" ]]
+}
+
+@test "does not support 'not equal' version constraint expressions" {
+  local -r tmpdir="$(mktemp -d)"
+  local -r version_file="${tmpdir}/main.tofu"
+  cat <<EOF >"${version_file}"
+opentofu {
+  required_version = "!= 0.13.7"
+}
+EOF
+
+  local -r actual_opentofu_version="$(ASDF_OPENTOFU_THIS_PLUGIN=opentofu "${PARSE_LEGACY_FILE}" "${version_file}")"
+
+  [[ ${actual_opentofu_version} == "" ]]
+}
+
+@test "does not support 'greater than' version constraint expressions" {
+  local -r tmpdir="$(mktemp -d)"
+  local -r version_file="${tmpdir}/main.tofu"
+  cat <<EOF >"${version_file}"
+opentofu {
+  required_version = "> 0.13.7"
+}
+EOF
+
+  local -r actual_opentofu_version="$(ASDF_OPENTOFU_THIS_PLUGIN=opentofu "${PARSE_LEGACY_FILE}" "${version_file}")"
+
+  [[ ${actual_opentofu_version} == "" ]]
+}
+
+@test "does not support 'less than' version constraint expressions" {
+  local -r tmpdir="$(mktemp -d)"
+  local -r version_file="${tmpdir}/main.tofu"
+  cat <<EOF >"${version_file}"
+opentofu {
+  required_version = "< 0.13.7"
+}
+EOF
+
+  local -r actual_opentofu_version="$(ASDF_OPENTOFU_THIS_PLUGIN=opentofu "${PARSE_LEGACY_FILE}" "${version_file}")"
+
+  [[ ${actual_opentofu_version} == "" ]]
+}
+
+@test "does not support squiggly arrow version constraint expressions" {
+  local -r tmpdir="$(mktemp -d)"
+  local -r version_file="${tmpdir}/main.tofu"
+  cat <<EOF >"${version_file}"
+opentofu {
+  required_version = "~> 0.13.7"
+}
+EOF
+
+  local -r actual_opentofu_version="$(ASDF_OPENTOFU_THIS_PLUGIN=opentofu "${PARSE_LEGACY_FILE}" "${version_file}")"
+
+  [[ ${actual_opentofu_version} == "" ]]
+}
+
+@test "does not support compound version constraint expressions" {
+  local -r tmpdir="$(mktemp -d)"
+  local -r version_file="${tmpdir}/main.tofu"
+  cat <<EOF >"${version_file}"
+opentofu {
+  required_version = "> 0.13.0, < 0.14.0"
+}
+EOF
+
+  local -r actual_opentofu_version="$(ASDF_OPENTOFU_THIS_PLUGIN=opentofu "${PARSE_LEGACY_FILE}" "${version_file}")"
+
+  [[ ${actual_opentofu_version} == "" ]]
+}
+
+# https://github.com/asdf-community/asdf-hashicorp/pull/43#discussion_r816027246
+@test 'does not get confused by multiple legacy version files for different plugins' {
+  local -r expected_opentofu_version=0.13.7
+  local -r tmpdir="$(mktemp -d)"
+  local -r version_file="${tmpdir}/main.tofu"
+  cat <<EOF >"${version_file}"
+opentofu {
+  required_version = "= ${expected_opentofu_version}"
+}
+EOF
+
+  echo 'foo' >"${tmpdir}/.packer-version"
+
+  local -r actual_opentofu_version="$(ASDF_OPENTOFU_THIS_PLUGIN=opentofu "${PARSE_LEGACY_FILE}" "${version_file}")"
+
+  [[ ${actual_opentofu_version} == "${expected_opentofu_version}" ]]
+}
+
```

Comment:
We do not have `.packer-version` in `list-legacy-filenames`.
Do you know if we need this test or if it can be removed?

Reply:
the hashicorp plugin supports legacy mode for terraform and packer, but in this plugin we can ignore packer. 
I changed the test so it creates an additional bogus .opentofu-version, and the tests still pass :heavy_check_mark: